code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm

```diff
@@ -105,6 +105,13 @@
 		but avoid actively interfering with the station, you are required to adminhelp and request permission to board the main station.</span>")
 	else
 		new_spawn.mind.store_memory("<b>Serve [owner.real_name], your creator.</b>")
+		if(owner.mind.special_role)
+			new_spawn.mind.store_memory("<b>[owner.real_name], your creator, is also an antagonist.</b>")
+			to_chat(new_spawn, "<b>[owner.real_name], your creator, is also an antagonist.</b>")
+			SSticker.mode.traitors.Add(new_spawn.mind)
+		if(locate(/datum/objective/hijack) in owner.mind.get_all_objectives())
+			new_spawn.mind.store_memory("<b>They must hijack the shuttle.</b>")
+			to_chat(new_spawn, "<b>They are also a hijacker.</b>")
```

Comment:
Any specific reason you didn't only have the second part of this proc run if they have a special role?
Also, for note, special roles != antagonist status *necessarily*, stuff like ERTs also have special roles, this will probably never come up but better safe than sorry if you're alright with adding a check for it